app/Services/CampaignService.php

```diff
@@ -176,4 +176,54 @@ public function getCampaignIdsFromSignups()
 
         return $ids ? $ids : null;
     }
+
+    /**
+     * Gets the count of pending, accepted, and rejected stautses on each post for a campaign.
+     *
+     * @param  array $campaigns
+     * @return Illuminate\Database\Eloquent\Collection $campaigns
+     */
+    public function getCampaignPostStatusCounts($campaigns)
+    {
+        $ids = array_filter(array_pluck($campaigns, 'id'));
+
+        $campaignsWithCounts = DB::table('signups')
+                ->leftJoin('posts', 'signups.id', '=', 'posts.signup_id')
+                ->select('signups.campaign_id',
+                    DB::raw('SUM(case when posts.status = "accepted" then 1 else 0 end) as accepted_count'),
+                    DB::raw('SUM(case when posts.status = "pending" then 1 else 0 end) as pending_count'),
+                    DB::raw('SUM(case when posts.status = "rejected" then 1 else 0 end) as rejected_count'))
+                ->wherein('campaign_id', $ids)
+                ->groupBy('signups.campaign_id')
+                ->get();
+
+        return $campaignsWithCounts ? collect($campaignsWithCounts)->keyBy('campaign_id') : null;
```

Comment:
Should this return an empty collection instead of `null`? Particularly thinking about what happens when we call a method on this expecting that it'll be a Collection instance down below:

```php
$statusCounts = $campaignsWithCounts->get($campaign['id']);
```